src/main/java/com/glencoesoftware/omero/ms/image/region/ShapeMaskRequestHandler.java

```diff
@@ -291,42 +291,218 @@ public ShapeMaskRequestHandler(ShapeMaskCtx shapeMaskCtx, String labelImagePath,
         }
     }
 
+    private int getCapacity(Domain domain) throws TileDBError {
+        long num_dims = domain.getNDim();
+        Datatype dimType = domain.getDimension(0).getType();
+        switch (dimType) {
+            case TILEDB_UINT8:
+            case TILEDB_INT8:{
+                int capacity = 1;
+                for(int i = 0; i < num_dims; i++) {
+                    byte start = (byte) (domain.getDimension(i).getDomain().getFirst());
```

Comment:
The type of the domain dimensions should always be the same. We are using `long` in `bioformats2raw` and should here as well. If the types are differing in the arrays then that is something you should double check with @erindiel.

Reply:
👍🏻 will update domain dimension type.